app/uk/gov/hmrc/gform/translation/TranslationController.scala

```diff
@@ -104,22 +90,32 @@ class TranslationController(
       ) { csv =>
         formTemplateService
           .get(FormTemplateRawId(formTemplateId.value))
-          .map(r => FormTemplateRaw(r.value + ("languages" -> Json.toJson(Seq("en", "cy")))))
-          .flatMap(t =>
-            handler.handleRequest(t).value.flatMap {
-              case Right(_)              => Future.successful(t)
-              case Left(unexpectedState) => Future.failed(new RuntimeException(unexpectedState.error))
-            }
-          )
+          .map(json => insertLanguages(json.value))
           .flatMap { json =>
-            val jsonAsString = Json.prettyPrint(json.value)
+            val jsonAsString = Json.prettyPrint(json)
             val jsonToSave = Json.parse(TextExtractor.translateFile(csv, jsonAsString)).as[JsObject]
             interpreter
               .handleRequest(FormTemplateRaw(jsonToSave))
               .fold(_.asBadRequest, _ => Ok(jsonToSave))
           }
       }
     }
+  private def insertLanguages(json: JsObject): JsObject = {
+    val fields = json.fields
+    val updatedFields = if (fields.exists { case (key, _) => key == "languages" }) {
+      fields.map {
+        case (key, value) if key == "languages" => (key, Json.toJson(Seq("en", "cy")))
+        case otherwise                          => otherwise
+      }
+    } else {
+      val (before, after) = fields.span(_._1 != "version")
+      after match {
+        case version :: as => before ++ Seq(version, ("languages", Json.toJson(Seq("en", "cy")))) ++ as
+        case _             => before ++ Seq(("languages", Json.toJson(Seq("en", "cy"))))
+      }
+    }
```

Comment:
Can we use `===` and `=!=` instead of `==` and `!=` to make ensure we are comparing same types?

Reply:
Also it would be nice to define `Json.toJson(Seq("en", "cy"))` just once

Reply:
done